contentcuration/contentcuration/frontend/shared/vuex/snackbar/index.js

```diff
@@ -19,7 +20,15 @@ export default {
   },
   actions: {
     showSnackbar({ commit }, { text, duration, actionText, actionCallback }) {
-      commit('CORE_CREATE_SNACKBAR', { text, duration, actionText, actionCallback });
+      return new Promise(hideCallback => {
```

Comment:
Could you give an example of where this is necessary? I feel like we should update the interaction flow instead of syncing the timing of other actions with the snackbar's timing

Reply:
So an example is when a snackbar has an action, like "UNDO". When the snackbar hits its duration limit, this would be resolved. This is useful for "cleanup" after that actionable time period the snackbar was present. I can avoid this if there's concern, but that would mean setting a matching timer or a separate expiry on the locks preventing syncing recent changes to the server.

Reply:
Ah nice! Yes, that should be okay for now- in the future, we'll hopefully have a more complete change history system so that we can decouple it from the snackbars